File: apps/web-asset/src/api/userSettings.ts
import { http } from "./http";

export type DisplayCurrency = "KRW" | "USD";

export type UserSettingsOut = {
  user_id: number;
  home_dashboard_id: number | null;
  default_scope_type: "USER" | "HOUSEHOLD";
  default_scope_id: number | null;
  hide_values: boolean;
  name_clamp_enabled: boolean;
  mobile_allocation_top_n: number;
  hide_small_assets: boolean;
  small_asset_threshold: string | number;
  display_currency: DisplayCurrency;
  created_at: string;
  updated_at: string;
};

export type UserSettingsUpdateIn = {
  display_currency?: DisplayCurrency;
  name_clamp_enabled?: boolean;
  mobile_allocation_top_n?: number;
};

export async function getMySettings(): Promise<UserSettingsOut> {
  const { data } = await http.get<UserSettingsOut>("/users/me/settings");
  return data;
}

export async function updateMySettings(payload: UserSettingsUpdateIn): Promise<UserSettingsOut> {
  const { data } = await http.patch<UserSettingsOut>("/users/me/settings", payload);
  return data;
}

File: apps/api/app/schemas/user_settings.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


DisplayCurrency = Literal["KRW", "USD"]


class UserSettingsOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    user_id: int
    home_dashboard_id: int | None
    default_scope_type: str
    default_scope_id: int | None
    hide_values: bool
    name_clamp_enabled: bool
    mobile_allocation_top_n: int
    hide_small_assets: bool
    small_asset_threshold: Decimal
    display_currency: DisplayCurrency
    created_at: datetime
    updated_at: datetime


class UserSettingsUpdate(BaseModel):
    display_currency: DisplayCurrency | None = None
    name_clamp_enabled: bool | None = None
    mobile_allocation_top_n: int | None = Field(default=None, ge=3, le=12)
